user-interface/src/main/java/life/qbic/datamanager/views/projects/project/samples/registration/batch/BatchRegistrationDialog2.java

```diff
@@ -0,0 +1,547 @@
+        .filter(
+            experimentalGroup -> experimentalGroup.condition().equals(matchingCondition))
+        .findAny();
+    // set experimental group in sampleInfo
+    matchingExperimentalGroup.ifPresent(sampleInfo::setExperimentalGroup);
+  }
+
+  private List<SampleInfo> prefilledSampleInfos() {
+    return prefilledSampleInfos(species, specimen, analytes, experimentalGroups);
+  }
+
+  private static List<SampleInfo> prefilledSampleInfos(List<Species> species,
+      List<Specimen> specimen, List<Analyte> analytes, List<ExperimentalGroup> experimentalGroups) {
+
+    List<SampleInfo> sampleInfos = new ArrayList<>();
+    for (ExperimentalGroup experimentalGroup : experimentalGroups) {
+      List<BiologicalReplicate> sortedReplicates = experimentalGroup.biologicalReplicates().stream()
+          .sorted(Comparator.comparing(BiologicalReplicate::label))
+          .toList();
+      for (BiologicalReplicate biologicalReplicate : sortedReplicates) {
+        // new sampleInfo
+        SampleInfo sampleInfo = new SampleInfo();
+        sampleInfo.biologicalReplicate = biologicalReplicate;
+        sampleInfo.experimentalGroup = experimentalGroup;
+        if (species.size() == 1) {
+          sampleInfo.setSpecies(species.get(0));
+        }
+        if (specimen.size() == 1) {
+          sampleInfo.setSpecimen(specimen.get(0));
+        }
+        if (analytes.size() == 1) {
+          sampleInfo.setAnalyte(analytes.get(0));
+        }
+        sampleInfos.add(sampleInfo);
+      }
+    }
+    return sampleInfos;
+  }
+
+  private static String formatConditionString(Condition condition) {
+    return condition.getVariableLevels().parallelStream()
+        .map(variableLevel -> "%s:%s %s"
+            .formatted(variableLevel.variableName().value(),
+                variableLevel.experimentalValue().value(),
+                variableLevel.experimentalValue().unit().orElse(""))
+            .trim())
+        .sorted()
+        .collect(Collectors.joining("; "));
+  }
+
+  private static ComponentRenderer<Span, AnalysisMethod> getAnalysisMethodItemRenderer() {
+    return new ComponentRenderer<>(analysisMethod -> {
+      var listItem = new Span();
+      listItem.addClassName("spreadsheet-list-item");
+      Span label = new Span(analysisMethod.label());
+      label.setText(analysisMethod.label());
+      var questionMarkIcon = VaadinIcon.QUESTION_CIRCLE_O.create();
+      questionMarkIcon.setTooltipText(analysisMethod.description());
+      listItem.add(label, questionMarkIcon);
+      return listItem;
+    });
+  }
+
+  private void onBatchNameChanged(
+      ComponentValueChangeEvent<TextField, String> batchNameChangedEvent) {
+    /* do nothing */
+  }
+
+  private void onRemoveLastRowClicked(ClickEvent<Button> clickEvent) {
+    spreadsheet.removeLastRow();
+  }
+
+  private void onAddRowClicked(ClickEvent<Button> clickEvent) {
+    spreadsheet.setValidationMode(ValidationMode.LAZY);
```

Comment:
I suspect this is a trick so the validation does not show errors if the row is added? 
```suggestion
    spreadsheet.setValidationMode(ValidationMode.LAZY);
```

Reply:
The validation is turned off, and on again after the new row is present 😄